comp/core/workloadmeta/collectors/internal/crio/containers.go

```diff
@@ -0,0 +1,298 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build crio
+
+// Package crio implements the crio Workloadmeta collector.
+package crio
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"time"
+
+	v1 "k8s.io/cri-api/pkg/apis/runtime/v1"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	"github.com/DataDog/datadog-agent/pkg/util/crio"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// convertToEvent converts a CRI-O container to a workloadmeta event.
+func (c *collector) convertToEvent(ctx context.Context, ctr *v1.Container) workloadmeta.CollectorEvent {
+	name := getContainerName(ctr.GetMetadata())
+	namespace := getPodNamespace(ctx, c.client, ctr.GetPodSandboxId())
+	containerStatus, info := getContainerStatus(ctx, c.client, ctr.GetId())
+	pid, hostname, cgroupsPath := parseContainerInfo(info)
+	cpuLimit, memLimit := getResourceLimits(containerStatus, info)
+	image := getContainerImage(ctx, c.client, ctr.GetImage())
+	ports := extractPortsFromAnnotations(ctr.GetAnnotations())
+
+	return workloadmeta.CollectorEvent{
+		Type:   workloadmeta.EventTypeSet,
+		Source: workloadmeta.SourceRuntime,
```

Comment:
[Please update the comment](https://github.com/DataDog/datadog-agent/blob/86dab7cf210a986cb2e5d8ea6497476444f4c6ed/comp/core/workloadmeta/def/types.go#L61-L64) for `SourceRuntime` to include `crio`